### ISSUE TITLE
Installation/usage on Windows 10?

### ISSUE THREAD

**Issue by nmanove** [jotyGill/openpyn-nordvpn#86]
Any documentation on this?

Thanks!

**Comment by brubrant** [comment]
Why not use the official client?

Em 10 de jan de 2018 16:58, "nmanove" <notifications@github.com> escreveu:

**Comment by nmanove** [comment]
Because I have no idea how to do that.

**Comment by nmanove** [comment]
It tells me to use sudo but sudo doesn't work on Windows 10, so I guess I need to use Ubuntu?

**Comment by nmanove** [comment]
And if I download it when I try to open setup.py nothing happens...

**Comment by jotyGill** [comment]
Instructions mate, they are important, you are the kind of guy/girl who skips game tutorials, aren't you? (we all do lol).
Check the **Installation Methods** Section on the main github page.

**Comment by brubrant** [comment]
This is a client for Nordvpn. If you are using Windows you can just
download their official "app" on their website.

Em 10 de jan de 2018 20:17, "nmanove" <notifications@github.com> escreveu:

**Comment by nmanove** [comment]
Thanks for the help, but I'm trying to run things on OpenVPN.
My latest issue is this:
sudo apt install openvpn python-gobject unzip wget
Reading package lists... Done
Building dependency tree
Reading state information... Done
Package python-gobject is not available, but is referred to by another package.
This may mean that the package is missing, has been obsoleted, or
is only available from another source
However the following packages replace it:
  python-gobject-2

E: Package 'python-gobject' has no installation candidate

This is where I'm stuck.

**Comment by nmanove** [comment]
So I've managed to install openpyn-nordvpn using Ubuntu.

But I'm getting this error after I enter my credentials:

Running openpyn for the first time? running 'openpyn --update' for you :)

--2018-01-11 18:55:13--  https://nordvpn.com/api/files/zip
Resolving nordvpn.com (nordvpn.com)... 104.20.16.34, 104.20.17.34
Connecting to nordvpn.com (nordvpn.com)|104.20.16.34|:443... connected.
HTTP request sent, awaiting response... 301 Moved Permanently
Location: https://downloads-legacy.nordvpn.com/configs/archives/servers/legacy/config.zip [following]
--2018-01-11 18:55:14--  https://downloads-legacy.nordvpn.com/configs/archives/servers/legacy/config.zip
Resolving downloads-legacy.nordvpn.com (downloads-legacy.nordvpn.com)... 104.20.16.34, 104.20.17.34
Connecting to downloads-legacy.nordvpn.com (downloads-legacy.nordvpn.com)|104.20.16.34|:443... connected.
HTTP request sent, awaiting response... 304 Not Modified
File ‘/usr/share/openpyn/zip’ not modified on server. Omitting download.

sudo: unzip: command not found
Exception occured while wgetting zip

--------------------------------------

Sorry for all the trouble...

**Comment by jotyGill** [comment]
I see they have changed the package name "python-gobject", but on ubuntu desktop it should be installed anyway.
"sudo: unzip: command not found":   Do you have unzip?

**Comment by nmanove** [comment]
Thanks for all your help. Sorry I'm so slow to catch on --- never really run linux or any shells before.

I've installed zip and things are moving--I've installed openpyn but here's an example of what I'm getting:

nmanove@DESKTOP-IRHD390:~$ sudo openpyn --init
Storing credentials in '/usr/share/openpyn/credentials with openvpn compatible 'auth-user-pass' file format
Awesome, the credentials have been saved in '/usr/share/openpyn/credentials'

Enter Openpyn options to be stored in systemd    service file (/etc/systemd/system/openpyn.service,     Default is, uk :uk

The Following config has been saved in openpyn.service. You can Run it or/and Enable it with: 'sudo systemctl start openpyn', 'sudo systemctl enable openpyn'

 [Unit]
Description=NordVPN connection manager
After=multi-user.target
[Service]
Type=simple
User=root
WorkingDirectory=/usr/share/openpyn/
ExecStart=/usr/bin/openpynuk --silent
StandardOutput=syslog
StandardError=syslog
[Install]
WantedBy=multi-user.target

Failed to connect to bus: No such file or directory
--2018-01-12 18:20:38--  https://nordvpn.com/api/files/zip
Resolving nordvpn.com (nordvpn.com)... 104.20.16.34, 104.20.17.34
Connecting to nordvpn.com (nordvpn.com)|104.20.16.34|:443... connected.
HTTP request sent, awaiting response... 301 Moved Permanently
Location: https://downloads-legacy.nordvpn.com/configs/archives/servers/legacy/config.zip [following]
--2018-01-12 18:20:39--  https://downloads-legacy.nordvpn.com/configs/archives/servers/legacy/config.zip
Resolving downloads-legacy.nordvpn.com (downloads-legacy.nordvpn.com)... 104.20.17.34, 104.20.16.34
Connecting to downloads-legacy.nordvpn.com (downloads-legacy.nordvpn.com)|104.20.17.34|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 11514233 (11M) [application/zip]
Saving to: ‘/usr/share/openpyn/zip’

zip                           100%[=================================================>]  10.98M  3.60MB/s    in 3.1s

2018-01-12 18:20:42 (3.60 MB/s) - ‘/usr/share/openpyn/zip’ saved [11514233/11514233]

Archive:  /usr/share/openpyn/zip
To see usage options type: "openpyn -h" or "openpyn --help"
nmanove@DESKTOP-IRHD390:~$ openpyn us -a 'new york'
According to NordVPN, Least Busy 4 Servers, In US in Location ['new york', 'new york county', 'new york', 'ny'] With 'Load' less than 70 Which Support OPENVPN-UDP are : [['us910', 19], ['us478', 20], ['us491', 20], ['us814', 20]]

Pinging Server us910 min/avg/max/mdev =  [124, 133, 151, 10]

Pinging Server us478 min/avg/max/mdev =  [17, 19, 22, 1]

Pinging Server us491 min/avg/max/mdev =  [19, 20, 23, 1]

Pinging Server us814 min/avg/max/mdev =  [19, 20, 21, 0]

Top 4 Servers with best Ping are: ['us478', 'us814', 'us491', 'us910']

Out of the Best Available Servers, Chose us478
CONNECTING TO SERVER us478 ON PORT udp1194
Your OS'Ubuntu' Does have '/sbin/resolvconf' using it to update DNS Resolver Entries

Fri Jan 12 18:22:08 2018 OpenVPN 2.3.10 x86_64-pc-linux-gnu [SSL (OpenSSL)] [LZO] [EPOLL] [PKCS11] [MH] [IPv6] built on Jun 22 2017
Fri Jan 12 18:22:08 2018 library versions: OpenSSL 1.0.2g  1 Mar 2016, LZO 2.08
Fri Jan 12 18:22:08 2018 MANAGEMENT: TCP Socket listening on [AF_INET]127.0.0.1:7015
Fri Jan 12 18:22:08 2018 WARNING: --ping should normally be used with --ping-restart or --ping-exit
Fri Jan 12 18:22:08 2018 NOTE: the current --script-security setting may allow this configuration to call user-defined scripts
Fri Jan 12 18:22:08 2018 Control Channel Authentication: tls-auth using INLINE static key file
Fri Jan 12 18:22:08 2018 Outgoing Control Channel Authentication: Using 160 bit message hash 'SHA1' for HMAC authentication
Fri Jan 12 18:22:08 2018 Incoming Control Channel Authentication: Using 160 bit message hash 'SHA1' for HMAC authentication
Fri Jan 12 18:22:08 2018 Socket Buffers: R=[163840->163840] S=[163840->163840]
Fri Jan 12 18:22:08 2018 UDPv4 link local: [undef]
Fri Jan 12 18:22:08 2018 UDPv4 link remote: [AF_INET]23.227.207.38:1194
Fri Jan 12 18:22:08 2018 TLS: Initial packet from [AF_INET]23.227.207.38:1194, sid=86dccebf 0a4e4449
Fri Jan 12 18:22:08 2018 WARNING: this configuration may cache passwords in memory -- use the auth-nocache option to prevent this
Fri Jan 12 18:22:08 2018 VERIFY OK: depth=1, C=PA, ST=PA, L=Panama, O=NordVPN, OU=NordVPN, CN=us478.nordvpn.com, name=NordVPN, emailAddress=cert@nordvpn.com
Fri Jan 12 18:22:08 2018 Validating certificate key usage
Fri Jan 12 18:22:08 2018 ++ Certificate has key usage  00a0, expects 00a0
Fri Jan 12 18:22:08 2018 VERIFY KU OK
Fri Jan 12 18:22:08 2018 Validating certificate extended key usage
Fri Jan 12 18:22:08 2018 ++ Certificate has EKU (str) TLS Web Server Authentication, expects TLS Web Server Authentication
Fri Jan 12 18:22:08 2018 VERIFY EKU OK
Fri Jan 12 18:22:08 2018 VERIFY OK: depth=0, C=PA, ST=PA, L=Panama, O=NordVPN, OU=NordVPN, CN=us478.nordvpn.com, name=NordVPN, emailAddress=cert@nordvpn.com
Fri Jan 12 18:22:08 2018 Data Channel Encrypt: Cipher 'AES-256-CBC' initialized with 256 bit key
Fri Jan 12 18:22:08 2018 Data Channel Encrypt: Using 160 bit message hash 'SHA1' for HMAC authentication
Fri Jan 12 18:22:08 2018 Data Channel Decrypt: Cipher 'AES-256-CBC' initialized with 256 bit key
Fri Jan 12 18:22:08 2018 Data Channel Decrypt: Using 160 bit message hash 'SHA1' for HMAC authentication
Fri Jan 12 18:22:08 2018 Control Channel: TLSv1.2, cipher TLSv1/SSLv3 ECDHE-RSA-AES256-GCM-SHA384, 2048 bit RSA
Fri Jan 12 18:22:08 2018 [us478.nordvpn.com] Peer Connection Initiated with [AF_INET]23.227.207.38:1194
Fri Jan 12 18:22:09 2018 MANAGEMENT: Client connected from [AF_INET]127.0.0.1:7015
Traceback (most recent call last):
  File "/usr/bin/openpyn-management", line 9, in <module>
    load_entry_point('openpyn==2.1.1', 'console_scripts', 'openpyn-management')()
  File "/usr/lib/python3/dist-packages/openpyn/management/management.py", line 34, in show
    notification.show()
GLib.Error: g-io-error-quark: Cannot autolaunch D-Bus without X11 $DISPLAY (0)
Fri Jan 12 18:22:09 2018 MANAGEMENT: Client disconnected
Fri Jan 12 18:22:11 2018 SENT CONTROL [us478.nordvpn.com]: 'PUSH_REQUEST' (status=1)
Fri Jan 12 18:22:11 2018 PUSH: Received control message: 'PUSH_REPLY,redirect-gateway def1,sndbuf 524288,rcvbuf 524288,dhcp-option DNS 78.46.223.24,dhcp-option DNS 162.242.211.137,route-gateway 10.8.8.1,topology subnet,ping 60,ping-restart 180,ifconfig 10.8.8.70 255.255.255.0,peer-id 42'
Fri Jan 12 18:22:11 2018 OPTIONS IMPORT: timers and/or timeouts modified
Fri Jan 12 18:22:11 2018 OPTIONS IMPORT: --sndbuf/--rcvbuf options modified
Fri Jan 12 18:22:11 2018 Socket Buffers: R=[163840->327680] S=[163840->327680]
Fri Jan 12 18:22:11 2018 OPTIONS IMPORT: --ifconfig/up options modified
Fri Jan 12 18:22:11 2018 OPTIONS IMPORT: route options modified
Fri Jan 12 18:22:11 2018 OPTIONS IMPORT: route-related options modified
Fri Jan 12 18:22:11 2018 OPTIONS IMPORT: --ip-win32 and/or --dhcp-option options modified
Fri Jan 12 18:22:11 2018 OPTIONS IMPORT: peer-id set
Fri Jan 12 18:22:11 2018 OPTIONS IMPORT: adjusting link_mtu to 1593
Fri Jan 12 18:22:11 2018 ROUTE: default_gateway=UNDEF
Fri Jan 12 18:22:11 2018 ERROR: Cannot open TUN/TAP dev /dev/net/tun: No such file or directory (errno=2)
Fri Jan 12 18:22:11 2018 Exiting due to fatal error
---------------------------------------------------------

Not sure what to do at this point.

**Comment by nmanove** [comment]
Sorry, don't know why the last lines are large and bolded...

**Comment by nmanove** [comment]
I also get this:

nmanove@DESKTOP-IRHD390:~$ openpyn us --area "new york"
Traceback (most recent call last):
  File "/usr/bin/openpyn", line 9, in <module>
    load_entry_point('openpyn==2.1.1', 'console_scripts', 'openpyn')()
  File "/usr/lib/python3/dist-packages/openpyn/openpyn.py", line 120, in main
    args.skip_dns_patch, args.silent)
  File "/usr/lib/python3/dist-packages/openpyn/openpyn.py", line 260, in run
    dedicated, double_vpn, tor_over_vpn, anti_ddos)
  File "/usr/lib/python3/dist-packages/openpyn/openpyn.py", line 355, in find_better_servers
    double_vpn=double_vpn, tor_over_vpn=tor_over_vpn, anti_ddos=anti_ddos)
  File "/usr/lib/python3/dist-packages/openpyn/openpyn.py", line 338, in get_data_from_api
    type_country_area_filtered = filters.filter_by_area(area, type_country_filtered)
  File "/usr/lib/python3/dist-packages/openpyn/filters.py", line 8, in filter_by_area
    resolved_locations = locations.get_unique_locations(list_of_servers=type_country_filtered)
  File "/usr/lib/python3/dist-packages/openpyn/locations.py", line 15, in get_unique_locations
    geo_address_list = get_location_name(eachLocation)
  File "/usr/lib/python3/dist-packages/openpyn/locations.py", line 36, in get_location_name
    results = r.json()['results'][0]['address_components']
IndexError: list index out of range

**Comment by jotyGill** [comment]
--area uses Google maps api, it was being blocked by google. That has been worked on (still only in test branch). But still you only get a fixed number of requests per day per IP.

If you are using in a NON-GUI os (Ubuntu Server), use the option --silent to keep the output clean from the error of management-client.

The error in "openvpn" is a weird one, beyond this program's scope.
Simply try and see:

`openpyn us --silent`

**Comment by jotyGill** [comment]
Hi nmanove,
If you still need help getting it working on GNU/Linux os's or Mac please open another issue. As openpyn is not supported on windows (the official client from NordVPN is available), I would like to close this issue. 

Cheers,

**Comment by nmanove** [comment]
Thanks!
